pysat/_instrument.py

```diff
@@ -555,6 +607,7 @@ def empty(self):
         if self.pandas_format:
             return self.data.empty
         else:
+            # TODO: What is 'time' is not one of the indexes?
```

Comment:
"What *if* 'time' is not one of the indices?" ?

Reply:
That is the big question.

The current pandas format as used by pysat requires time for the main DataFrame index. In that sense, this is not anything new. If the xarray straight from the file doesn't have time as the label, it would be up to the load routine to make it so.

While we are here, should 'time' be 'epoch' instead?

I'm not trying to enforce pandas paradigms on xarray, but keeping the time assumption helps ensure the rest of pysat continues to work with little modification. The experience I have with testing the xarray pysat source is that it seemed to require that I put in 'time' specifically when I coupled it to the pysat data access and setting mechanisms. If I don't have to then great.

The trouble comes in because xarray is super general and there can be things in an xarray that are not indexed by time. I'm currently not sure what to do about this, I need a specific case to deal with. I say that based on the history of pysat... pysat is the way it is because the problems I ran into forced rethinking previously help assumptions which then generally forced evolution of the code.

Reply:
So, there is probably a very good reason not to require time as the index.  However, I also think that if you want to work together with observational and modelled data then you will need to specify time.  So, in this context, it makes sense to require time as the index for modelled data.  This isn't as hard as it sounds, since you know the start time, run time, and time step of the simulations.

Also, I think the name is good.  'epoch' begs the question "which one?" whilst "time" implies datetime in a python context.

Reply:
Also, my original comment was a grammatical correction :-P